library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -313,9 +321,111 @@ PlatformBridgeFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
   if (status == Http::FilterTrailersStatus::StopIteration) {
     pending_response_trailers_ = &trailers;
   }
+  response_complete_ = true;
   return status;
 }
 
+void PlatformBridgeFilter::onResumeDecoding() {
+  Buffer::Instance* internal_buffer = nullptr;
+  if (decoder_callbacks_->decodingBuffer()) {
+    decoder_callbacks_->modifyDecodingBuffer([&internal_buffer](Buffer::Instance& mutable_buffer) {
+      internal_buffer = &mutable_buffer;
+    });
+  }
+
+  envoy_headers* pending_headers = nullptr;
+  envoy_data* pending_data = nullptr;
+  envoy_headers* pending_trailers = nullptr;
+  if (pending_request_headers_) {
+    *pending_headers = Http::Utility::toBridgeHeaders(*pending_request_headers_);
+  }
+  if (internal_buffer) {
+    *pending_data = Buffer::Utility::copyToBridgeData(*internal_buffer);
+  }
+  if (pending_request_trailers_) {
+    *pending_trailers = Http::Utility::toBridgeHeaders(*pending_request_trailers_);
+  }
+
+  envoy_filter_resume_status result =
+      platform_filter_.on_resume_request(pending_headers, pending_data, pending_trailers,
+                                         request_complete_, platform_filter_.instance_context);
+  if (result.status == kEnvoyFilterResumeStatusStopIteration) {
+    return;
+  }
+  if (pending_request_headers_) {
+    PlatformBridgeFilter::replaceHeaders(*pending_request_headers_, *result.pending_headers);
+    pending_request_headers_ = nullptr;
+    free(result.pending_headers);
+  }
+  if (internal_buffer) {
+    internal_buffer->drain(internal_buffer->length());
+    internal_buffer->addBufferFragment(
+        *Buffer::BridgeFragment::createBridgeFragment(*result.pending_data));
+    free(result.pending_data);
+  } else if (result.pending_data) {
+    Buffer::OwnedImpl inject_data;
+    inject_data.addBufferFragment(
+        *Buffer::BridgeFragment::createBridgeFragment(*result.pending_data));
+    decoder_callbacks_->addDecodedData(inject_data, false);
```

Comment:
is `false` referring to `end_stream` here? If so, is it safe to always assume `false` (how does the consumer close the stream?)?

Reply:
The `false` actually is part of an unrelated Envoy API that has to do with whether buffer watermarking will occur - which has to do with backpressure, and isn't applicable for a library. (The documentation on the API is actually a little bit ambiguous to me, but `false` seemed correct based on my reading of it and other usage I've encountered.)

Reply:
Got it. Let's do an inline `/* watermark */` to annotate